### ISSUE TITLE
[JENKINS-74986] errorprone parser not detecting issues

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/analysis-model#1422]
<p>Having the following log output in jenkins the error prone parser does not detect warnings anymore</p>

<p> </p>

```
11:24:08  [WARNING] Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8 -Djansi.force=true -Dmaven.ext.class.path="/foofoo_master@​tmp/withMaven88890a94/pipeline-maven-spy.jar" -Dorg.jenkinsci.plugins.pipeline.maven.reportsFolder="/foofoo_master@​tmp/withMaven88890a94" 
11:24:08  /foofoo/app/src/main/java/foofoo/MonitorRepository.java:[29,7] [MissingOverride] deleteById implements method in CrudRepository; expected @​Override
11:24:08      (see https://errorprone.info/bugpattern/MissingOverride)
11:24:08    Did you mean '@​Override @​Modifying(flushAutomatically = true, clearAutomatically = true)'?
11:24:08  [INFO] 
11:24:08  [INFO] --- resources:3.3.1:testResources (default-testResources) @ app ---
11:24:08  [INFO] Copying 6 resources from src/test/resources to target/test-classes
11:24:08  [INFO] 
11:24:08  [INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ app ---
11:24:08  [INFO] Recompiling the module because of changed dependency.
11:24:08  [INFO] Compiling 58 source files with javac [forked debug parameters release 21] to target/test-classes
11:24:16  [WARNING] Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8 -Djansi.force=true -Dmaven.ext.class.path="/foofoo_master@​tmp/withMaven88890a94/pipeline-maven-spy.jar" -Dorg.jenkinsci.plugins.pipeline.maven.reportsFolder="/foofoo_master@​tmp/withMaven88890a94" 
11:24:16  /foofoo/app/src/test/java/foofoo/DeviceTestDataGenerator.java:[54,4] [BadImport] Importing nested classes/static methods/static fields with commonly-used names can make code harder to read, because it may not be clear from the context exactly which type is being referred to. Qualifying the name with that of the containing class can make the code clearer. Here we recommend using qualified class: FooModel.
11:24:16      (see https://errorprone.info/bugpattern/BadImport)
11:24:16    Did you mean 'FooModel.Builder builder = FooModel.newBuilder()'?
11:24:16  [INFO] 
11:24:16  [INFO] --- jacoco:0.8.12:prepare-agent (pre-unit-test) @ app ---
11:24:16  [INFO] surefireArgLine set to -javaagent:/home/jenkins/agent/workspace/BA-RD-CIT_app-oneview_master/.repository/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/foofoo_master/bar/app/target/coverage-reports/jacoco-ut.exec,excludes=**/*_.*
11:24:16  [INFO] 
```
<p>Looks like that the JAVA TOOLS OPTION warning somehow interfers with the logging</p>

<p>Maybe its also that we run errorProne as a separate maven step with the default compile just annotation processing and not compilation</p>

<pre>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <executions>
          <execution>
            <id><span class="code-keyword">default</span>-compile</id>
            <phase>generate-sources</phase>
            <goals>
              <goal>compile</goal>
            </goals>
            <configuration>
              <skipMain>${skipAP}</skipMain>
              <compilerArgs>
                <arg>-implicit:<span class="code-keyword">class&</span>amp;lt;/arg>
                <arg>-XprintProcessorInfo</arg>
                <arg>-XprintRounds</arg>
              </compilerArgs>
              <proc>only</proc>
            </configuration>
          </execution>
          <execution>
            <id>errorprone-compile</id>
            <phase>compile</phase>
            <goals>
              <goal>compile</goal>
            </goals>
            <configuration>
              <fork><span class="code-keyword">true</span></fork>
              <annotationProcessorPaths>
                <path>
                  <groupId>com.google.errorprone</groupId>
                  <artifactId>error_prone_core</artifactId>
                  <version>${version.error-prone}</version>
                </path>
              </annotationProcessorPaths>
              <compilerArgs combine.children=<span class="code-quote">"append"</span>>
                <arg>-XDcompilePolicy=simple</arg>
                <arg>--should-stop=ifError=FLOW</arg>
                <arg>-Xplugin:ErrorProne ${error-prone.flags}</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED</arg>
                <arg>-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED</arg>
                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED</arg>
                <arg>-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED</arg>
              </compilerArgs>
              <generatedSourcesDirectory>${project.build.directory}/generated-sources/unused</generatedSourcesDirectory>
            </configuration>
          </execution> 
        </executions>
      </plugin></pre>
<p> </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=reitzmichnicht">reitzmichnicht</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-74986" target="_blank">errorprone parser not detecting issues</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: analysis-model
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2025-04-11T13:53:29+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 20260108-221235
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Having the following log output in jenkins the error prone parser does not detect warnings anymore</p>

<p> </p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>11:24:08  [WARNING] Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8 -Djansi.force=true -Dmaven.ext.class.path="/foofoo_master@tmp/withMaven88890a94/pipeline-maven-spy.jar" -Dorg.jenkinsci.plugins.pipeline.maven.reportsFolder="/foofoo_master@tmp/withMaven88890a94" 
11:24:08  /foofoo/app/src/main/java/foofoo/MonitorRepository.java:[29,7] [MissingOverride] deleteById implements method in CrudRepository; expected @Override
11:24:08      (see https://errorprone.info/bugpattern/MissingOverride)
11:24:08    Did you mean '@Override @Modifying(flushAutomatically = true, clearAutomatically = true)'?
11:24:08  [INFO] 
11:24:08  [INFO] --- resources:3.3.1:testResources (default-testResources) @ app ---
11:24:08  [INFO] Copying 6 resources from src/test/resources to target/test-classes
11:24:08  [INFO] 
11:24:08  [INFO] --- compiler:3.13.0:testCompile (default-testCompile) @ app ---
11:24:08  [INFO] Recompiling the module because of changed dependency.
11:24:08  [INFO] Compiling 58 source files with javac [forked debug parameters release 21] to target/test-classes
11:24:16  [WARNING] Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8 -Djansi.force=true -Dmaven.ext.class.path="/foofoo_master@tmp/withMaven88890a94/pipeline-maven-spy.jar" -Dorg.jenkinsci.plugins.pipeline.maven.reportsFolder="/foofoo_master@tmp/withMaven88890a94" 
11:24:16  /foofoo/app/src/test/java/foofoo/DeviceTestDataGenerator.java:[54,4] [BadImport] Importing nested classes/static methods/static fields with commonly-used names can make code harder to read, because it may not be clear from the context exactly which type is being referred to. Qualifying the name with that of the containing class can make the code clearer. Here we recommend using qualified class: FooModel.
11:24:16      (see https://errorprone.info/bugpattern/BadImport)
11:24:16    Did you mean 'FooModel.Builder builder = FooModel.newBuilder()'?
11:24:16  [INFO] 
11:24:16  [INFO] --- jacoco:0.8.12:prepare-agent (pre-unit-test) @ app ---
11:24:16  [INFO] surefireArgLine set to -javaagent:/home/jenkins/agent/workspace/BA-RD-CIT_app-oneview_master/.repository/org/jacoco/org.jacoco.agent/0.8.12/org.jacoco.agent-0.8.12-runtime.jar=destfile=/foofoo_master/bar/app/target/coverage-reports/jacoco-ut.exec,excludes=**/*_.*
11:24:16  [INFO] </pre>
</div></div>
<p>Looks like that the JAVA TOOLS OPTION warning somehow interfers with the logging</p>

<p>Maybe its also that we run errorProne as a separate maven step with the default compile just annotation processing and not compilation</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">      &lt;plugin&gt;
        &lt;groupId&gt;org.apache.maven.plugins&lt;/groupId&gt;
        &lt;artifactId&gt;maven-compiler-plugin&lt;/artifactId&gt;
        &lt;executions&gt;
          &lt;execution&gt;
            &lt;id&gt;<span class="code-keyword">default</span>-compile&lt;/id&gt;&lt;!-- runs only annotation processors, but need to override <span class="code-keyword">default</span> id <span class="code-keyword">for</span> intellij to recognize compiler args --&gt;
            &lt;phase&gt;generate-sources&lt;/phase&gt;
            &lt;goals&gt;
              &lt;goal&gt;compile&lt;/goal&gt;
            &lt;/goals&gt;
            &lt;configuration&gt;
              &lt;skipMain&gt;${skipAP}&lt;/skipMain&gt;
              &lt;compilerArgs&gt;
                &lt;arg&gt;-implicit:<span class="code-keyword">class&</span>amp;lt;/arg&gt;&lt;!-- set explicit <span class="code-keyword">for</span> annotation processors --&gt;
                &lt;arg&gt;-XprintProcessorInfo&lt;/arg&gt;&lt;!-- Print information about which annotations a processor is asked to process --&gt;
                &lt;arg&gt;-XprintRounds&lt;/arg&gt;&lt;!-- Print information about initial and subsequent annotation processing rounds --&gt;
              &lt;/compilerArgs&gt;
              &lt;proc&gt;only&lt;/proc&gt;&lt;!-- runs only annotation processors --&gt;
            &lt;/configuration&gt;
          &lt;/execution&gt;
          &lt;execution&gt;
            &lt;id&gt;errorprone-compile&lt;/id&gt;
            &lt;phase&gt;compile&lt;/phase&gt;
            &lt;goals&gt;
              &lt;goal&gt;compile&lt;/goal&gt;
            &lt;/goals&gt;
            &lt;configuration&gt;
              &lt;fork&gt;<span class="code-keyword">true</span>&lt;/fork&gt;
              &lt;annotationProcessorPaths&gt;
                &lt;path&gt;
                  &lt;groupId&gt;com.google.errorprone&lt;/groupId&gt;
                  &lt;artifactId&gt;error_prone_core&lt;/artifactId&gt;
                  &lt;version&gt;${version.error-prone}&lt;/version&gt;
                &lt;/path&gt;
              &lt;/annotationProcessorPaths&gt;
              &lt;compilerArgs combine.children=<span class="code-quote">"append"</span>&gt;
                &lt;arg&gt;-XDcompilePolicy=simple&lt;/arg&gt;&lt;!-- recommended by error-prone --&gt;
                &lt;arg&gt;--should-stop=ifError=FLOW&lt;/arg&gt;
                &lt;arg&gt;-Xplugin:ErrorProne ${error-prone.flags}&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED&lt;/arg&gt;
                &lt;arg&gt;-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED&lt;/arg&gt;
              &lt;/compilerArgs&gt;
              &lt;generatedSourcesDirectory&gt;${project.build.directory}/generated-sources/unused&lt;/generatedSourcesDirectory&gt;&lt;!-- otherwise it ignores sources already available --&gt;
            &lt;/configuration&gt;
          &lt;/execution&gt; 
        &lt;/executions&gt;
      &lt;/plugin&gt;</pre>
</div></div>
<p> </p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 2.479.1<br/>
Analysis Model API Plugin 12.9.1<br/>
Maven 3.9.x<br/>
JDK 21
```
</details>
